Source/ACE/Command/Handlers/AccountCommands.cs

```diff
@@ -11,19 +11,128 @@ namespace ACE.Command.Handlers
+            string salt             = SHA2.Hash(SHA2Type.SHA256, Path.GetRandomFileName());
+            string password         = SHA2.Hash(SHA2Type.SHA256, parameters[1]);
+            AccessLevel accessLevel = AccessLevel.Player;
+
+            if (parameters.Length > 2)
+            {
+                switch (parameters[2].ToLower())
+                {
+                    case "1":
+                    case "advocate":
+                        accessLevel = AccessLevel.Advocate;
+                        break;
+                    case "2":
+                    case "sentinel":
+                        accessLevel = AccessLevel.Sentinel;
+                        break;
+                    case "3":
+                    case "envoy":
+                        accessLevel = AccessLevel.Envoy;
+                        break;
+                    case "4":
+                    case "developer":
+                        accessLevel = AccessLevel.Developer;
+                        break;
+                    case "5":
+                    case "admin":
+                        accessLevel = AccessLevel.Admin;
+                        break;
+                    default:
+                        accessLevel = AccessLevel.Player;
+                        break;
+                }
+            }
+
+            string articleAorAN = "a";
+            if (accessLevel == AccessLevel.Advocate || accessLevel == AccessLevel.Admin || accessLevel == AccessLevel.Envoy)
+                articleAorAN = "an";
+
+            Account acc = new Account(accountId, account, accessLevel, salt, password);
 
             DatabaseManager.Authentication.CreateAccount(acc);
 
-            Console.WriteLine("Account successfully created for " + account + ".");
+            Console.WriteLine("Account successfully created for " + account + " with access rights as " + articleAorAN + " " + Enum.GetName(typeof(AccessLevel), accessLevel) + ".");
+        }
+
+        // set-accountaccess accountname (accesslevel)
+        [CommandHandler("set-accountaccess", AccessLevel.Admin, CommandHandlerFlag.None, 1)]
+        public static void HandleAccountUpdateAccessLevel(Session session, params string[] parameters)
+        {
+            uint accountId      = 0;
+            string accountName  = parameters[0].ToLower();
+
+            try
+            {
+                DatabaseManager.Authentication.GetAccountIdByName(accountName, out accountId);
+            }
+            catch (IndexOutOfRangeException)
+            {
+                if (session == null)
+                    Console.WriteLine("Account " + accountName + " does not exist.");
+                else
+                    ChatPacket.SendSystemMessage(session, "Account " + accountName + " does not exist.");
+                return;
+            }
+
+            AccessLevel accessLevel = AccessLevel.Player;
+
+            if (parameters.Length > 1)
+            {
+                switch (parameters[1].ToLower())
```

Comment:
again, Enum.Parse should do here.  No need to do a messy switch (and less maintenance).